apps/guardian/src-tauri/src/main.rs

```diff
@@ -1029,7 +1046,363 @@ async fn generate_secret_share_flow<'a>(
         };
 
         // Wait for response from the user through the frontend
-        wait_for_approval(&channel_id, &window, proposal).await?;
+        wait_for_approval(&channel_id, &window, &proposal).await?;
+
+        contract_update
+            .send(&active_guardian.guardian.keys)
+            .await?
+            .wait_for_finalization()
+            .await?;
+
+        Ok(())
+    };
+
+    tokio::select! {
+        error = cancel => Err(error),
+        res = interaction => res
+    }
```

Comment:
Maybe add `biased` here, towards the `cancel` future?

Reply:
Makes sense, didn't know this was a thing!